executor/executor.cc

```diff
@@ -423,23 +436,21 @@ int main(int argc, char** argv)
 #if SYZ_EXECUTOR_USES_SHMEM
 	if (mmap(&input_data[0], kMaxInput, PROT_READ, MAP_PRIVATE | MAP_FIXED, kInFd, 0) != &input_data[0])
 		fail("mmap of input file failed");
-	// The output region is the only thing in executor process for which consistency matters.
-	// If it is corrupted ipc package will fail to parse its contents and panic.
-	// But fuzzer constantly invents new ways of how to currupt the region,
-	// so we map the region at a (hopefully) hard to guess address with random offset,
-	// surrounded by unmapped pages.
-	// The address chosen must also work on 32-bit kernels with 1GB user address space.
-	void* preferred = (void*)(0x1b2bc20000ull + (1 << 20) * (getpid() % 128));
-	output_data = (uint32*)mmap(preferred, kMaxOutput,
-				    PROT_READ | PROT_WRITE, MAP_SHARED | MAP_FIXED, kOutFd, 0);
-	if (output_data != preferred)
-		fail("mmap of output file failed");
 
 	// Prevent test programs to mess with these fds.
 	// Due to races in collider mode, a program can e.g. ftruncate one of these fds,
 	// which will cause fuzzer to crash.
 	close(kInFd);
+
+#ifdef SYZ_EXECUTOR_USES_FORK_SERVER
+	// First allocate a minimal amount of memory. The children will allocate the rest, if needed.
+	mmap_output(kMaxOutputBasic);
+	// close(kOutFd) is invoked in the child, after the program has been received.
```

Comment:
I think it's better to not close it here in both modes and always close in the child. Then we can just remove some ifdef's and make the logic mode uniform across modes (always better because we test one mode much better than the other).

Reply:
> not close it here in both modes and always close in the child

But there's no child in the case of `ifndef SYZ_EXECUTOR_USES_FORK_SERVER`, so we would anyway have to handle it separately - in one case make sure that we're closing it only one time, in the other case we need to close those handles before executing each program because it's done in forked children.